src/rgen/Microsoft.Macios.Generator/DataModel/MemberCodeChange.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.Macios.Generator.DataModel;
+
+/// <summary>
+/// Structure that represents a change that was made by the user on a members that has to be
+/// reflected in the generated code.
+/// </summary>
+readonly struct MemberCodeChange : IEquatable<MemberCodeChange> {
+
+	/// <summary>
+	/// Get the name of the member.
+	/// </summary>
+	public string Name { get; }
+
+	/// <summary>
+	/// Get the attributes added to the member.
+	/// </summary>
+	public ImmutableArray<AttributeCodeChange> Attributes { get; }
+
+	/// <summary>
+	/// Create a new change that happened on a member.
+	/// </summary>
+	/// <param name="name">The name of the changed member.</param>
+	/// <param name="attributes">The list of attribute changes in the member.</param>
+	public MemberCodeChange (string name, ImmutableArray<AttributeCodeChange> attributes)
+	{
+		Name = name;
+		Attributes = attributes;
+	}
+
+	/// <summary>
+	/// Create a new change that happened on a member.
+	/// </summary>
+	/// <param name="name">The name of the changed member.</param>
+	public MemberCodeChange (string name) : this (name, []) { }
+
+	/// <inheritdoc />
+	public bool Equals (MemberCodeChange other)
+	{
+		if (Name != other.Name || Attributes.Length != other.Attributes.Length)
+			return false;
+		// sort the attributes to make sure the order is the same and compare them
+		var xAttributes = Attributes.OrderBy (s => s.Name).ToArray ();
+		var yAttributes = other.Attributes.OrderBy (s => s.Name).ToArray ();
+		for (var index = 0; index < xAttributes.Length; index++) {
+			if (xAttributes [index] != yAttributes [index])
+				return false;
+		}
```

Comment:
Maybe add a helper method for comparing attributes? This is the second time it's done in this PR, and this implementation has a bug: there will be an IndexOutOfRange exception if there are more `Attributes` than `other.Attributes`.

Reply:
The IndexOutOfRange should be handled by the line https://github.com/xamarin/xamarin-macios/pull/21517/files/5b333806aadc48710e9649d7474eb21f9fd0ed5b#diff-c21e4853b55b3e54f9a8e3e71340c5c531c1d7e8d777cf55e4360189bef24eb8R43

I can add a comparer for AttributeLists, that not a bad idea.

Reply:
Added a test to show it https://github.com/xamarin/xamarin-macios/pull/21517/commits/d98a87283a2eba498a4056b32a49383103841b13 

I'll add the function in a few

Reply:
There's another bug, these two won't compare identically:

```cs
[SupportedOSPlatform ("ios15.0")]
[SupportedOSPlatform ("maccatalyst15.0")]
```

```cs
[SupportedOSPlatform ("maccatalyst15.0")]
[SupportedOSPlatform ("ios15.0")]
```

because you sort first based on the name only.

Reply:
Good catch, nothing that an IComparer cannot solve, added it plus a test: https://github.com/xamarin/xamarin-macios/pull/21517/commits/4f27eac4d7ce4a9d645916d074b8746dfafa39df